builtin/logical/pki/path_config_acme_test.go

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package pki
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestAcmeConfig(t *testing.T) {
+	cluster, client, _ := setupAcmeBackend(t)
+	defer cluster.Cleanup()
+
+	cases := []struct {
+		name        string
+		AcmeConfig  map[string]interface{}
+		prefixUrl   string
+		validConfig bool
+		works       bool
+	}{
+		{"unspecified-root", map[string]interface{}{
+			"enabled":         true,
+			"allowed_issuers": "*",
+			"allowed_roles":   "*",
+			"dns_resolver":    "",
+			"eab_policy_name": "",
+		}, "acme/", true, true},
+		{"bad-policy-root", map[string]interface{}{
+			"enabled":                  true,
+			"allowed_issuers":          "*",
+			"allowed_roles":            "*",
+			"default_directory_policy": "bad",
+			"dns_resolver":             "",
+			"eab_policy_name":          "",
+		}, "acme/", false, false},
+		{"forbid-root", map[string]interface{}{
+			"enabled":                  true,
+			"allowed_issuers":          "*",
+			"allowed_roles":            "*",
+			"default_directory_policy": "forbid",
+			"dns_resolver":             "",
+			"eab_policy_name":          "",
+		}, "acme/", true, false},
+		{"sign-verbatim-root", map[string]interface{}{
+			"enabled":                  true,
+			"allowed_issuers":          "*",
+			"allowed_roles":            "*",
+			"default_directory_policy": "sign-verbatim",
+			"dns_resolver":             "",
+			"eab_policy_name":          "",
+		}, "acme/", true, true},
+		{"role-root", map[string]interface{}{
+			"enabled":                  true,
+			"allowed_issuers":          "*",
+			"allowed_roles":            "*",
+			"default_directory_policy": "role:exists",
+			"dns_resolver":             "",
+			"eab_policy_name":          "",
+		}, "acme/", true, true},
+		{"bad-role-root", map[string]interface{}{
+			"enabled":                  true,
+			"allowed_issuers":          "*",
+			"allowed_roles":            "*",
+			"default_directory_policy": "role:notgood",
+			"dns_resolver":             "",
+			"eab_policy_name":          "",
+		}, "acme/", false, true},
+		{"disallowed-role-root", map[string]interface{}{
+			"enabled":                  true,
+			"allowed_issuers":          "*",
+			"allowed_roles":            "good",
+			"default_directory_policy": "role:exists",
+			"dns_resolver":             "",
+			"eab_policy_name":          "",
+		}, "acme/", false, false},
+	}
+
+	roleConfig := map[string]interface{}{
+		"issuer_ref":       "default",
+		"allowed_domains":  "example.com",
+		"allow_subdomains": true,
+		"max_ttl":          "720h",
+	}
+
+	testCtx := context.Background()
+
+	for _, tc := range cases {
+		deadline := time.Now().Add(time.Second * 10)
+		subTestCtx, _ := context.WithDeadline(testCtx, deadline)
```

Comment:
Really not worth changing, just an fyi that this could be simplified with, WithTimeout calls WithDeadline with the time.Now().Add for you.

```
subTestCtx, _ := context.WithTimeout(testCtx, 10*time.Second)
```